src/prpl.ts

```diff
@@ -173,17 +181,30 @@ self.addEventListener('activate', () => self.registration.unregister());`);
       // We handle the caching header ourselves.
       cacheControl: false,
     };
+    send(request, fileToSend, sendOpts)
+        .on('error',
+            (err: httpErrors.HttpError) => {
+              // `send` puts a lot of detail in the error message, like the
+              // absolute system path of the missing file for a 404. We don't
+              // want that to leak out, so let's use a generic message instead.
+              err.message = statuses[err.status] || String(err.status);
+              handleError(err);
+            })
+        .pipe(response);
+  };
+}
 
-    let stream = send(request, fileToSend, sendOpts);
 
-    // Set a custom error-handling function
-    if (config && config.error) {
-      const errorFunction = (error:  httpErrors.HttpError) => config.error!(request, response, error)
-      stream = stream.on('error', errorFunction);
-    }
-
-    stream.pipe(response);
-  };
+/**
+ * Write a plain text HTTP error response.
+ */
+function writePlainTextError(
+    response: http.ServerResponse, error: httpErrors.HttpError) {
+  response.statusCode = error.status;
+  response.setHeader('Content-Type', 'text/plain');
+  response.setHeader(
+      'Content-Length', Buffer.byteLength(error.message).toString());
```

Comment:
This goes through Express right? Pretty sure Express sets this for you, fine to leave it though I guess.

Reply:
It will not necessarily go through Express (the tests actually just test a vanilla node HTTP server). However, it looks like node's HTTP server automatically adds content-length too. Removed this line, and added checks in the tests for it instead.